diff --git a/Makefile b/Makefile
index 7c52acc..1c7a955 100644
--- a/Makefile
+++ b/Makefile
@@ -135,8 +135,10 @@ world: prepare $(target/stamp-compile) $(package/stamp-compile) $(package/stamp-
 	$(_SINGLE)$(SUBMAKE) -r json_overview_image_info
 	$(_SINGLE)$(SUBMAKE) -r checksum
 ifneq ($(CONFIG_CCACHE),)
+ifneq ($(CONFIG_CCACHE),sccache)
 	$(STAGING_DIR_HOST)/bin/ccache -s
 endif
+endif
 
 .PHONY: clean dirclean prereq prepare world package/symlinks package/symlinks-install package/symlinks-clean
 
diff --git a/include/cmake.mk b/include/cmake.mk
index f59410c..a884c9e 100644
--- a/include/cmake.mk
+++ b/include/cmake.mk
@@ -36,7 +36,11 @@ ifeq ($(CONFIG_CCACHE),)
  CMAKE_HOST_C_COMPILER:=$(HOSTCC)
  CMAKE_HOST_CXX_COMPILER:=$(HOSTCXX)
 else
+  ifeq ($(CONFIG_CCACHE),sccache)
+  CCACHE:=sccache
+  else
   CCACHE:=$(STAGING_DIR_HOST)/bin/ccache
+  endif
   CMAKE_C_COMPILER_LAUNCHER:=$(CCACHE)
   CMAKE_CXX_COMPILER_LAUNCHER:=$(CCACHE)
   CMAKE_C_COMPILER:=$(TARGET_CC_NOCACHE)
diff --git a/rules.mk b/rules.mk
index 54df407..0351438 100644
--- a/rules.mk
+++ b/rules.mk
@@ -338,12 +338,19 @@ export HOSTCC_NOCACHE
 export HOSTCXX_NOCACHE
 
 ifneq ($(CONFIG_CCACHE),)
-  TARGET_CC:= ccache $(TARGET_CC)
-  TARGET_CXX:= ccache $(TARGET_CXX)
-  HOSTCC:= ccache $(HOSTCC)
-  HOSTCXX:= ccache $(HOSTCXX)
-  export CCACHE_BASEDIR:=$(TOPDIR)
-  export CCACHE_DIR:=$(if $(call qstrip,$(CONFIG_CCACHE_DIR)),$(call qstrip,$(CONFIG_CCACHE_DIR)),$(TOPDIR)/.ccache)
+  ifeq ($(CONFIG_CCACHE),sccache)
+    TARGET_CC:= sccache $(TARGET_CC)
+    TARGET_CXX:= sccache $(TARGET_CXX)
+    HOSTCC:= sccache $(HOSTCC)
+    HOSTCXX:= sccache $(HOSTCXX)
+  else
+    TARGET_CC:= ccache $(TARGET_CC)
+    TARGET_CXX:= ccache $(TARGET_CXX)
+    HOSTCC:= ccache $(HOSTCC)
+    HOSTCXX:= ccache $(HOSTCXX)
+    export CCACHE_BASEDIR:=$(TOPDIR)
+    export CCACHE_DIR:=$(if $(call qstrip,$(CONFIG_CCACHE_DIR)),$(call qstrip,$(CONFIG_CCACHE_DIR)),$(TOPDIR)/.ccache)
+  endif
   export CCACHE_COMPILERCHECK:=%compiler% -dumpmachine; %compiler% -dumpversion
 endif
 
diff --git a/tools/Makefile b/tools/Makefile
index b16c5d9..2659f9d 100644
--- a/tools/Makefile
+++ b/tools/Makefile
@@ -146,9 +146,11 @@ endif
 
 ifneq ($(CONFIG_CCACHE)$(CONFIG_SDK),)
+ifneq ($(CONFIG_CCACHE),sccache)
 $(foreach tool, $(filter-out zstd zlib xz pkgconf patch ninja meson libressl expat cmake,$(tools-y)), $(eval $(curdir)/$(tool)/compile += $(curdir)/ccache/compile))
 tools-y += ccache
 $(curdir)/ccache/compile := $(curdir)/cmake/compile
 endif
+endif
 
 # in case there is no patch tool on the host we need to make patch tool a
 # dependency for tools which have patches directory
